framework/web/CHttpSessionHandler.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * CHttpSessionHandler class file.
+ *
+ * @author Qiang Xue <qiang.xue@gmail.com>
+ * @link https://www.yiiframework.com/
+ * @copyright 2008-2013 Yii Software LLC
+ * @license https://www.yiiframework.com/license/
+ */
+
+/**
+ * SessionHandlerInterface adapter for CHttpSession.
+ * Delegates all calls to CHttpSession's openSession/closeSession/etc. methods,
+ * allowing subclasses like CDbHttpSession to work without modification.
+ */
+class CHttpSessionHandler implements SessionHandlerInterface
+{
+	/**
+	 * @var CHttpSession
+	 */
+	private $_session;
+
+	/**
+	 * @param CHttpSession $session
+	 */
+	public function __construct(CHttpSession $session)
+	{
+		$this->_session=$session;
+	}
+
+	/**
+	 * @param string $path
+	 * @param string $name
+	 * @return bool
+	 */
+	#[ReturnTypeWillChange]
+	public function open($path, $name)
+	{
+		return $this->_session->openSession($path, $name);
+	}
+
+	/**
+	 * @return bool
+	 */
+	#[ReturnTypeWillChange]
+	public function close()
+	{
+		return $this->_session->closeSession();
+	}
+
+	/**
+	 * @param string $id
+	 * @return string|false
+	 */
+	#[ReturnTypeWillChange]
+	public function read($id)
+	{
+		return $this->_session->readSession($id);
+	}
+
+	/**
+	 * @param string $id
+	 * @param string $data
+	 * @return bool
+	 */
+	#[ReturnTypeWillChange]
+	public function write($id, $data)
+	{
+		return $this->_session->writeSession($id, $data);
+	}
+
+	/**
+	 * @param string $id
+	 * @return bool
+	 */
+	#[ReturnTypeWillChange]
+	public function destroy($id)
+	{
+		return $this->_session->destroySession($id);
+	}
+
+	/**
+	 * @param int $max_lifetime
+	 * @return int|false
+	 */
+	#[ReturnTypeWillChange]
+	public function gc($max_lifetime)
+	{
+		return $this->_session->gcSession($max_lifetime) ? 0 : false;
```

Comment:
The [documentation on the interface](https://www.php.net/manual/en/sessionhandlerinterface.gc.php) says:

> Returns the number of deleted sessions on success, or [false](https://www.php.net/manual/en/reserved.constants.php#constant.false) on failure. Note this value is returned internally to PHP for processing.

Since Yii doesn't count this number your implementation always returns int 0 on success. Do you expect any possible problems to arise from that? Maybe we could take a quick look in what way PHP wants to use this value for internal processing.